hummingbot/hummingbot
# Fixes infinite loop in start_mqtt_async - hummingbot/brokers/issues/8

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:

Fixes errors produced while using the `mqtt_autostart`. Reported in: hummingbot/brokers/issues/8#issuecomment-1846971660


**Tests performed by the developer**:

Tested 2 Cases.

Case 1 - `mqtt_autostart` enabled on startup:

- Broker is disconnected
- Start the bot
- Import a strategy
- Start the strategy
- Start the broker
- Bot should connect successfully and without any disconnection logs

Case 2 - `mqtt_autostart` disabled on startup:

- Broker is disconnected
- Start the bot
- Set the `mqtt_autostart` to True
- Connect to broker
- Import a strategy
- Start the strategy
- Start the broker
- Bot should connect successfully and without any disconnection logs

And of course I have tested while the broker is running before starting the bot

**Tips for QA testing**:
